cpu/esp_common/esp-riscv/exceptions.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp_common
+ * @{
+ *
+ * @file
+ * @brief       Exception handling for RISC-V-based ESP SoCs
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @}
+ */
+
+#include <inttypes.h>
+
+#include "kernel_defines.h"
+#include "periph/pm.h"
+
+#include "esp_attr.h"
+#include "riscv/rvruntime-frames.h"
+#include "rom/ets_sys.h"
+
+static const char *exceptions[] = {
+    "nil",
+    "0x1: PMP Instruction access fault",
+    "0x2: Illegal Instruction",
+    "0x3: Hardware Breakpoint/Watchpoint or EBREAK",
+    "nil",
+    "0x5: PMP Load access fault",
+    "nil",
+    "0x7: PMP Store access fault",
+    "0x8: ECALL from U mode",
+    "nil",
+    "nil",
+    "0xb: ECALL from M mode",
+};
+
+void init_exceptions (void)
+{
+}
+
+void IRAM_ATTR xt_unhandled_exception(RvExcFrame *frame)
+{
+    /* TODO */
+    ets_printf("#! Unhandled exception @0x%08"PRIx32", cause %s, "
+               "powering off\n", frame->mepc, exceptions[frame->mcause]);
+    if (!IS_USED(MODULE_ESP_GDB)) {
+        pm_off();
+    }
+    while (1) { };
```

Comment:
Why not call

```suggestion
    core_panic(PANIC_DUMMY_HANDLER, "unhandled exception");
```

instead?

Reply:
> Why not call

The implementation just corresponds to that for Xtensa. The initial intention was that the system stops in the endless while loop to be able to attach the debugger after a crash provided that the app has been compiled with debugging information (module `esp_gdb`). Otherwise the system is powered off. `core_panic` powers off the system if `DEVELHELP` is defined.

I don' remember exactly, but I'm pretty sure I have seen a similar approach with the exception of the `esp_gdb` module in another CPU implementation when I was porting RIOT to ESPs, including the very specific message `!# ...`.

We can change it to whatever makes sense. However, a platform implementation `PANIC_DUMMY_HANDLER` only exists on Cortex-M platforms. Maybe `PANIC_UNDEFINED` or `PANIC_UNDEFINED` could be used. Furthermore, it would make sense to have a `panic_arch` function.

BTW, in Xtensa implementation of `panic_arch` https://github.com/RIOT-OS/RIOT/blob/dd574a45f2465c220c77c4669e29847be2d68991/cpu/esp_common/esp-xtensa/exceptions.c#L199-L206 the system watchdog timer is used to reset the system if `DEVELHELP` is defined and the `break` instruction was executed while there is no debugger attached.